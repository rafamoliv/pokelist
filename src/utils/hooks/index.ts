import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { RootState, AppDispatch } from '@/store'
import { useLocalStorage } from './uselLocalStorage'

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export { useLocalStorage }
