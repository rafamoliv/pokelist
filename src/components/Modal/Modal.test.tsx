import i18next from 'i18next'
import { render, screen } from '@/utils/tests'
import { Modal } from '.'
import { ReactNode } from 'react'

const props = {
  children: i18next.t('translation:foo'),
  loading: true,
  isOpen: true,
  onClose: vi.mock('')
}

describe('<Modal />', () => {
  describe('Component Snapshot', () => {
    it('Render full element', () => {
      const { container } = render(
        <Modal.Root {...props}>
          <Modal.Body>Body</Modal.Body>
          <Modal.Footer>Footer</Modal.Footer>
        </Modal.Root>
      )
      expect(container.firstChild).toMatchSnapshot()
    })

    it('Render with just Body element', () => {
      const { container } = render(
        <Modal.Root {...props}>
          <Modal.Body>Body</Modal.Body>
        </Modal.Root>
      )
      expect(container.firstChild).toMatchSnapshot()
    })

    it('Render with just Footer element', () => {
      const { container } = render(
        <Modal.Root {...props}>
          <Modal.Footer>Footer</Modal.Footer>
        </Modal.Root>
      )
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Component Props', () => {
    it('Should render loading prop', () => {
      render(
        <Modal.Root isOpen={true} loading={true}>
          <Modal.Body>Body</Modal.Body>
          <Modal.Footer>Footer</Modal.Footer>
        </Modal.Root>
      )
      expect(screen.getByLabelText(`animation`)).toBeInTheDocument()
    })

    it('Should render Body children', () => {
      const { container } = render(
        <Modal.Root isOpen={true} loading={true}>
          <Modal.Body>Body</Modal.Body>
        </Modal.Root>
      )
      expect(container.firstChild).toBeTruthy()
    })

    it('Should render Footer children', () => {
      const { container } = render(
        <Modal.Root isOpen={true} loading={true}>
          <Modal.Footer>Footer</Modal.Footer>
        </Modal.Root>
      )
      expect(container.firstChild).toBeTruthy()
    })
  })
})
