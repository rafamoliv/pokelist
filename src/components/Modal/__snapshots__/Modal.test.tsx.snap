// Vitest Snapshot v1

exports[`<Modal /> > Component Snapshot > Render full element 1`] = `null`;

exports[`<Modal /> > Component Snapshot > Render with just Body element 1`] = `null`;

exports[`<Modal /> > Component Snapshot > Render with just Footer element 1`] = `null`;
